### PR TITLE
Idms dml Add support for SCHEMA and MAP sections #704

### PR DESCRIPTION
**Pull request by schte01**

### CODE REVIEW

File: server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4 [hunk 1]
```diff
@@ -941,7 +941,7 @@ cobolWord
    | ODT | ORDERLY | OWN | OF
    | PASSWORD | PORT | PRINTER | PROCESS | PROGRAM
    | READER | REAL | RECEIVED | RECURSIVE | REF | REMOTE | REMOVE
-   | SAVE | SHARED | SHORT_DATE | SQL | SYMBOL
+   | SAVE | SCHEMA | SHARED | SHORT_DATE | SQL | SYMBOL
```
Comment (temanbrcom) [inline server/src/main/antlr4/com/broadcom/lsp/cobol/core/parser/CICSParser.g4:944]: Seems like MAX and VERSION are missing. Is it intentional?

File: server/src/main/resources/com/broadcom/lsp/cobol/service/delegates/completions/LanguageKeywords.txt [hunk 2]
```diff
@@ -973,6 +974,7 @@ RUN=When STOP RUN is specified, execution is terminated and control is returned
 SADDRLENGTH=
 SAME=The SAME keyword has several functions.<br>[Search IBM documentation](https://www.ibm.com/support/knowledgecenter/SS6SG3_6.2.0/welcome.html)<br>\u00A9 Copyright IBM Corporation 1994, 2019.<br>U.S. Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 SAVE=
+SCHEMA=
```
Comment (temanbrcom) [inline server/src/main/resources/com/broadcom/lsp/cobol/service/delegates/completions/LanguageKeywords.txt:977]: VERSION is missing

File: server/src/test/java/com/broadcom/lsp/cobol/usecases/TestIdmsSections.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * These test for variations other idms SECTION definitions
+ */
+class TestIdmsSections {
+
+    private static final String MESSAGE_1 = "rule ss_names_length failed predicate: {_input.LT(1).getText().matches(\"16|18\")}?";
```
Comment (temanbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/usecases/TestIdmsSections.java:34]: Here and further the lines are not aligned properly

File: server/src/test/java/com/broadcom/lsp/cobol/usecases/TestIdmsSections.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    private static Stream<String> textsToTest() {
+        return Stream.of(
+                BOILERPLATE + IDMSSS_WITH_ALL_CLAUSES, BOILERPLATE + IDMSSS_NO_CS_NO_VERSION, BOILERPLATE + IDMSMS_WITH_ALL_CLAUSES);
+    }
+
+    @ParameterizedTest
+    @MethodSource("textsToTest")
+    @DisplayName("Parameterized - varying tests")
+    void test(String text) {
+        UseCaseEngine.runTest(text, List.of(), Map.of("1",
```
Comment (temanbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/usecases/TestIdmsSections.java:77]: I don't see this id used in the text example

**Review by temanbrcom — CHANGES_REQUESTED** [review]
I would recommend squashing those commits

**Review by temanbrcom — APPROVED** [review]
🍪

File: server/src/test/java/com/broadcom/lsp/cobol/usecases/TestIdmsSections.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          + "        MAP MAP3 TYPE STANDARD.\r\n";
+
+  private static Stream<String> textsToTest() {
+    return Stream.of(
+        BOILERPLATE + IDMSSS_WITH_ALL_CLAUSES,
+        BOILERPLATE + IDMSSS_NO_CS_NO_VERSION,
+        BOILERPLATE + IDMSMS_WITH_ALL_CLAUSES);
+  }
+
+  void test(String text) {
```
Comment (grianbrcom) [inline server/src/test/java/com/broadcom/lsp/cobol/usecases/TestIdmsSections.java:69]: This looks like a parametrized test. Can you, please, add those annotations:
```
  @ParameterizedTest
  @MethodSource("textsToTest")
```

**Review by grianbrcom — CHANGES_REQUESTED** [review]
Please, annotate the test. Otherwise, it is not run.

**Review by temanbrcom — APPROVED** [review]

**Review by grianbrcom — COMMENTED** [review]
:safety_pin:

**Review by grianbrcom — APPROVED** [review]

**Pull request merged**